risc0/zkvm/src/prove/mod.rs

```diff
@@ -185,20 +239,48 @@ impl<'a> Prover<'a> {
         })
     }
 
+    /// Provide input data to the guest. This data can be read by the guest
+    /// via [crate::guest::env::read].
+    ///
+    /// It is possible to provide multiple inputs to the guest so long as the
+    /// guest reads them in the same order they are added by the [Prover].
+    /// However, to reduce maintenance burden and the chance of mistakes, we
+    /// recommend instead using a single `struct` to hold all the inputs and
+    /// calling [Prover::add_input_u8_slice] just once (on that input
+    /// `struct`).
```

Comment:
```suggestion
    /// calling [Prover::add_input_u8_slice] just once (on the serialized representation of the input.
```

Reply:
One thing to consider is adding helper functions on the `Prover` which makes it easier to pass serialized inputs. Similar to how the guest has `read` vs `read_slice`. We could have `add_input_u8_slice` for the raw version, and `add_input` which would be generalized over a `Serializer`.